File: livingdoc-documentation/src/main/java/implementations/Calculator.java
package implementations;

/**
 * A simple calculator, just as in the sample but here written in JAVA
 *
 * This will be our first system under test (SUT) in this project
 */

public class Calculator {

    /**
     * @param a the dividend
     * @param b the divisor
     * @return the quotient of two numbers
     */

    public double sum(double a, double b) {
        return a + b;
    }

    /**
     * @param a the minuend
     * @param b the subtrahend
     * @return the difference of two numbers
     */

    public double diff(double a, double b) {
        return a - b;
    }

}

File: livingdoc-tests/src/main/kotlin/org/livingdoc/example/Calculator.kt
package org.livingdoc.example

/**
 * A simple calculator implementation
 *
 * This will be our first system under test (SUT) in this project
 */
class Calculator {

    /**
     * @return the sum of two numbers
     */
    fun sum(a: Float, b: Float): Float {
        return a + b
    }

    /**
     * @param a the minuend
     * @param b the subtrahend
     * @return the difference of two numbers
     */
    fun diff(a: Float, b: Float): Float {
        return a - b
    }

    /**
     * @return the product of two numbers
     */
    fun multiply(a: Float, b: Float): Float {
        return a * b
    }

    /**
     * @param a the dividend
     * @param b the divisor
     * @return the quotient of two numbers
     */
    fun divide(a: Float, b: Float): Float {
        return a / b
    }
}
